### ISSUE TITLE
[REVIEW]: Disimpy: A massively parallel Monte Carlo simulator for synthesizing diffusion-weighted MRI data in Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2527]
**Submitting author:** @kerkelae (<a href="http://orcid.org/0000-0001-9446-4305">Leevi Kerkelä</a>)
**Repository:** <a href="https://github.com/kerkelae/disimpy" target ="_blank">https://github.com/kerkelae/disimpy</a>
**Version:** v0.1.1
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** @DARSakthi, @grlee77, @ritagnunes
**Archive:** <a href="https://doi.org/10.5281/zenodo.4001687" target="_blank">10.5281/zenodo.4001687</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441/status.svg)](https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441"><img src="https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441/status.svg)](https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/kerkelae/disimpy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@DARSakthi & @grlee77 & @ritagnunes, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Kevin-Mattheus-Moerman know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kerkelae/disimpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kerkelae) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kerkelae/disimpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kerkelae) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kerkelae/disimpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kerkelae) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @DARSakthi, @grlee77, @ritagnunes it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.1007/978-3-642-15745-5_50 may be missing for title: High-fidelity meshes from tissue samples for diffusion MRI simulations
- https://doi.org/10.1016/j.jmr.2012.10.015 may be missing for title: Isotropic diffusion weighting in PGSE NMR by magic-angle spinning of the q-vector
- https://doi.org/10.1063/1.1695690 may be missing for title: Spin diffusion measurements: spin echoes in the presence of a time-dependent field gradient
- https://doi.org/10.1109/mcse.2011.37 may be missing for title: The NumPy array: a structure for efficient numerical computation
- https://doi.org/10.1109/tmi.2009.2015756 may be missing for title: Convergence and parameter choice for Monte-Carlo simulations of diffusion MRI
- https://doi.org/10.1016/j.neuroimage.2007.02.016 may be missing for title: Robust determination of the fibre orientation distribution in diffusion MRI: non-negativity constrained super-resolved spherical deconvolution
- https://doi.org/10.1002/nbm.3998 may be missing for title: Quantifying brain microstructure with diffusion MRI: Theory and parameter estimation
- https://doi.org/10.1101/140459 may be missing for title: The role of diffusion MRI in neuroscience
- https://doi.org/10.1109/hotchips.2008.7476516 may be missing for title: Scalable parallel programming with CUDA

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02527/joss.02527/10.21105.joss.02527.pdf)

**Comment by kerkelae** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02527/joss.02527/10.21105.joss.02527.pdf)

**Comment by kerkelae** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-3-642-15745-5_50 is OK
- 10.1016/j.jmr.2015.10.012 is OK
- 10.1016/j.jmr.2012.10.015 is OK
- 10.1063/1.1695690 is OK
- 10.1145/2833157.2833162 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1109/mcse.2011.37 is OK
- 10.1109/tmi.2009.2015756 is OK
- 10.1016/j.neuroimage.2007.02.016 is OK
- 10.1002/nbm.3998 is OK
- 10.1101/140459 is OK
- 10.1109/hotchips.2008.7476516 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@DARSakthi, @grlee77, @ritagnunes this is just a friendly check in. I know some of you are busy on other reviews too. Could you give an update on when you expect to work on this review?

**Comment by ritagnunes** [comment]
@Kevin-Mattheus-Moerman, most likely I will only be able to work on this
after the end of the ISMRM conference, i.e. the week after next...

**Comment by Kevin-Mattheus-Moerman** [comment]
Right, sorry I forgot about the ISMRM. Enjoy the conference!

**Comment by grlee77** [comment]
@kerkelae, can you briefly describe the contributions of the coauthors your included? I could only verify your name directly in the commit history, but understand that there are other ways to contribute.

**Comment by DARSakthi** [comment]
@Kevin-Mattheus-Moerman Thanks for the reminder -- some other deadlines had dragged my attention away -- will have it done by Monday, latest. Cheers

**Comment by grlee77** [comment]
regarding the following review criteria:
> If there are any performance claims of the software, have they been confirmed? 

Is the script that produced Fig. 1 available anywhere? I find the result believable, but did not try to install Camino to verify. I can confirm that all of the examples in the Jupyter notebook in the repository ran quickly on an NVIDIA GTX-1080 Ti card. 
@Kevin-Mattheus-Moerman: Is this adequate verification for JOSS?

**Comment by grlee77** [comment]
Overall the manuscript and software look good to me and I have only some minor suggestions. 

**minor points to address**

1.) In the NumPy reference, the first author should be "van der Walt, S." rather than "Walt, S. van der"

2.) In the second sentence under features, I recommend the following grammatical change:
"generated in massively parallel on" -> "generated in a massively parallel fashion on"

3.) In the documentation's installation instructions a one-step command that might be easier for users than downloading the repository and then navigating to the folder to run pip would be to use:
```pip install git+https://github.com/kerkelae/disimpy.git```
(pip will take care of downloading the repository for the user)

4.) It may be worth providing a version of the demo notebook on [Google Colaboratory](https://colab.research.google.com/notebooks/intro.ipynb) as well, so users can try it out without having to own a CUDA-compatible GPU.

5.) Can the authors confirm if Camino's CPU-based Monte-Carlo is multi-threaded? If so, I would mention that explicitly in the manuscript text or Fig. 1 caption as it makes the relative improvement on the GPU more impressive.

6.) Regarding the "state of the filed" review criteria, I don't see much in the manuscript corresponding to this aside from the comparison to Camino. This is not an area I am particular familiar with the literature, so it would be good to know:
-  is Monte-Carlo is the dominant method of doing these simulations or are there other popular approaches?
- are there other popular tools aside from Camino for doing this?

7.) The included references look good, but I would include also reference previous work doing similar types of simulations on the GPU. For example:
```
Khieu-Van Nguyen, Edwin Hernández-Garzón, Julien Valette, Efficient GPU-based Monte-Carlo simulation of diffusion in real astrocytes reconstructed from confocal microscopy, Journal of Magnetic Resonance, Volume 296, 2018, Pages 188-199, https://doi.org/10.1016/j.jmr.2018.09.013.

Christopher A. Waudby, John Christodoulou, GPU accelerated Monte Carlo simulation of pulsed-field gradient NMR experiments, Journal of Magnetic Resonance, Volume 211, Issue 1, 2011, Pages 67-73, https://doi.org/10.1016/j.jmr.2011.04.004.
```
(optional)
It may also be worth mentioning other complementary software that could generate more realistic tissue models as input to the simulations. I am not familiar with specific tools for this, but am aware of the following publication (unfortunately the implementation described in it does not appear to be publicly available):

```
Kévin Ginsburger, Felix Matuschke, Fabrice Poupon, Jean-François Mangin, Markus Axer, Cyril Poupon,
MEDUSA: A GPU-based tool to create realistic phantoms of the brain microstructure using tiny spheres,
NeuroImage, Volume 193, 2019, Pages 10-24, https://doi.org/10.1016/j.neuroimage.2019.02.055
```

**misc comments**
Not having continuous integration seems okay for this library as the free CI services do not offer the needed GPU hardware.

Validation of accuracy of the simulations was not discussed in the manuscript, but from what I can tell, the authors have done validation via the following:
1.) examples in the tutorials:
        - free water diffusion matches the analytical solution
        - diffraction peaks generated as in Avram et. al. 2008
2.) via tests/test_simulations.py
        - some test cases compare to stored outputs of Camino
        - mesh-based sphere gives similar result to an analytical sphere

**Comment by fnery** [comment]
Hi @grlee77, thanks for the feedback. Just a quick comment in terms of direct commits to the repo itself I didn't make a lot of them and they are no longer visible because we cleaned up the commit history for submission (made an orphan branch). They are still visible as [closed PRs](https://github.com/kerkelae/disimpy/pulls?q=is%3Apr+is%3Aclosed). In any case I will let @kerkelae clarify the describe authors' contributions.

**Comment by kerkelae** [comment]
> @kerkelae, can you briefly describe the contributions of the coauthors you included? I could only verify your name directly in the commit history, but understand that there are other ways to contribute.

Good point @grlee77. I am the main author and have written most of the code and of the manuscript. Fabio Nery has helped by implementing various algorithms, making decisions on how to package the code, and testing. Matt Hall and Chris Clark have contributed to the software by being very involved in the planning and development phase of much of the code that the final package consists of.

**Comment by DARSakthi** [comment]
Overall the paper is well done. The tutorial section of the documentation is a nice touch, and the mathematical specification in the final section of the paper is also appreciated. Having a colab notebook would be an excellent idea. 

I echo the suggestions of @grlee77, in particular verification of the performance claims. I do question the use of the word synthesising in the title -- while 'to synthesise' can mean to produce artificially, a more common usage is to combine. If the authors are partial to the title then no change is necessary, but a better word may clarify the function of the software.

**Comment by kerkelae** [comment]
@grlee77 and @DARSakthi, thank you for the feedback. We have made a number of changes to the manuscript and the documentation. Please see the points below.

- We made the small fixes and changes suggested by @grlee77. We also have no particular preference for the word "synthesize" which we replaced by "generate". The updated manuscript contains the changed title but I am not sure how to change the title of my JOSS submission. Could @Kevin-Mattheus-Moerman help us with this please?

- Please find [here](https://github.com/openjournals/joss-reviews/files/5061338/simple_comparison.zip) a simple script and a ply file that can be used to reproduce Figure 1. Since we didn't find large differences between runtimes of individual simulations, we reported a single set of runtimes instead of averages. Please note that reproducing the figure requires the installation of Java and Camino. Camino is single-threaded and many studies using Camino have run simulations in parallel on clusters.

- We simplified the installation instructions according to the suggestion by @grlee77 and expanded the tutorial to describe how to run the examples on Google Colaboratory. Special thanks for this tip!

- We expanded the "Statement of need" section to answer the points 6 and 7 by @grlee77. There exists other methods for simulating dMRI experiments, for example, by solving the Bloch-Torrey equation (which adds the diffusion term to the Bloch equation), but the Monte Carlo method is more generalizable. To our knowledge, Camino is the most popular Monte Carlo simulator but other simulators have been recently developed, some of which are GPU-accelerated. However, we are not aware of approachable Python packages for performing GPU-accelerated dMRI simulations which we find to be of value. We also mentioned the need for relevant microstructural models.

- Indeed, we do not have continuous integration since the services we are familiar with do not provide support for CUDA. To validate the functionality of the package, we have written unit tests and compared the simulated signals to analytical solutions and to Camino's results.

**Comment by kerkelae** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02527/joss.02527/10.21105.joss.02527.pdf)

**Comment by DARSakthi** [comment]
Is the test script you linked to available in the repo too? I can't seem to find it. If not, I think it could be a good thing to include from an end - user stand point, so as to evaluate whether to use the software or not. 

With the most recent round of changes, I am happy to recommend the paper for publication. Well done!

**Comment by ritagnunes** [comment]
I really enjoyed being able to run the Tutorial on Google Colaboratory. That was a really nice addition!
It will be very helpful for potential/new users to have a quick grasp of the implemented software features.

Also, the provided examples make it easy to verify the requirement of simulating a high number of random walkers to match the theoretical predictions motivating the need for this software.  I also like it that the authors have included a range of microstructural environments, from free to restricted diffusion including multiple compartments and a diffraction example replicating a previous study. 

It was harder for me to evaluate the gains in simulation speed as the software and corresponding tutorial do not run without GPU-enabled acceleration. Although I did not run Camino independently, I trust that the simulation times reported in the tests are correct.

The provided documentation is quite comprehensive and the paper clearly motivates the need for the software, describing the implemented features. I also really like that the basic theoretical equations have been included. 

I agree with @DARSakthi and I am also happy to recommend the publication of this paper.
Congratulations!

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev can you help advise as to how to change the title of a submission?

**Comment by Kevin-Mattheus-Moerman** [comment]
@grlee77 thanks again for your help. Can you review the changes implemented by the authors. 
You have two boxes unticked (state of the field, references), can you comment/summarize to the authors what is needed for these to be checked?

**Comment by arfon** [comment]
> @openjournals/dev can you help advise as to how to change the title of a submission?

Just change the title of the paper in the `paper.md` file.

**Comment by grlee77** [comment]
> @grlee77 thanks again for your help. Can you review the changes implemented by the authors.
You have two boxes unticked (state of the field, references), can you comment/summarize to the authors what is needed for these to be checked?

These were addressed by the revised statement of need. I have now checked them off and am satisfied with the publication.

**Comment by kerkelae** [comment]
Thank you to all the reviewers!

@arfon and @Kevin-Mattheus-Moerman, I have changed the title in the `paper.md` file. I just wanted to know if there is something more I need to do because the old title was still shown on the [JOSS website](https://joss.theoj.org/papers/ee3aa136859cb47595de7de8eb84b441)

**Comment by arfon** [comment]
> Thanks @arfon and @Kevin-Mattheus-Moerman. I had changed the title in the paper.md file and just wanted to know if there is something more I need to do because the old title was still shown on the JOSS website.

The version on the JOSS site will update when the paper is published.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02527/joss.02527/10.21105.joss.02527.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1016/j.neuroimage.2020.117107 is OK
- 10.1016/j.jmr.2018.09.013 is OK
- 10.3389/fninf.2020.00008 is OK
- 10.1016/j.neuroimage.2019.116120 is OK
- 10.1007/978-3-319-46630-9_4 is OK
- 10.1007/978-3-642-15745-5_50 is OK
- 10.1016/j.jmr.2015.10.012 is OK
- 10.1016/j.jmr.2012.10.015 is OK
- 10.1063/1.1695690 is OK
- 10.1145/2833157.2833162 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1109/mcse.2011.37 is OK
- 10.1109/tmi.2009.2015756 is OK
- 10.1016/j.neuroimage.2007.02.016 is OK
- 10.1002/nbm.3998 is OK
- 10.1101/140459 is OK
- 10.1109/hotchips.2008.7476516 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@kerkelae I've checked your paper and can confirm it looks in order. 

These are final steps to complete: 

- [ ] Please archive a copy of the software on ZENODO and report back here with the DOI of the archived version. When you create this archived version be sure to have the title and author meta data match that of the paper (you may need to manually alter this). 

- [ ] Can you inform me of the most recent version tag for this work? Is it still at `v0.1`?

**Comment by kerkelae** [comment]
@Kevin-Mattheus-Moerman 

Since we changed a few things during the review, I created a [new release](https://github.com/kerkelae/disimpy/releases/tag/v0.1.1): `v0.1.1`. I archived this version on [Zenodo](https://zenodo.org/record/4001687). The DOI is http://doi.org/10.5281/zenodo.4001687.

**Comment by Kevin-Mattheus-Moerman** [comment]
@kerkelae thanks. However you still need to amend the title of that archive to match the paper title. Let me know when you've completed that.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo.4001687 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4001687" target="_blank">10.5281/zenodo.4001687</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set v0.1.1 as version

**Comment by whedon** [comment]
OK. v0.1.1 is the version.

**Comment by kerkelae** [comment]
@Kevin-Mattheus-Moerman It's fixed now.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1016/j.neuroimage.2020.117107 is OK
- 10.1016/j.jmr.2018.09.013 is OK
- 10.3389/fninf.2020.00008 is OK
- 10.1016/j.neuroimage.2019.116120 is OK
- 10.1007/978-3-319-46630-9_4 is OK
- 10.1007/978-3-642-15745-5_50 is OK
- 10.1016/j.jmr.2015.10.012 is OK
- 10.1016/j.jmr.2012.10.015 is OK
- 10.1063/1.1695690 is OK
- 10.1145/2833157.2833162 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1109/mcse.2011.37 is OK
- 10.1109/tmi.2009.2015756 is OK
- 10.1016/j.neuroimage.2007.02.016 is OK
- 10.1002/nbm.3998 is OK
- 10.1101/140459 is OK
- 10.1109/hotchips.2008.7476516 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1673

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1673, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1298598917542678528) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1674
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02527](https://doi.org/10.21105/joss.02527)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02527/status.svg)](https://doi.org/10.21105/joss.02527)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02527">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02527/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02527/status.svg
   :target: https://doi.org/10.21105/joss.02527
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02527/status.svg)](https://doi.org/10.21105/joss.02527)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
